source/ceylon/logging/LoggerFactory.ceylon

```diff
@@ -0,0 +1,90 @@
+import ceylon.language.meta.model {
+    ClassOrInterface
+}
+import ceylon.collection {
+    MutableMap,
+    HashMap
+}
+
+"`LoggerFactory` can be satisfied to define a custom
+ logging strategy. Typically, an instance of `LoggerFactory`
+ will be be assigned to [[loggerFactory]]."
+shared
+interface LoggerFactory {
+
+    "Function to obtain a [[Logger]] for the given
+     [[Category]].
+
+     Callers intending to wrap the returned logger should
+     provide the [[ClassOrInterface]] of the wrapper.
+     For example:
+
+            FormattingLogger formattingLogger(Category category)
+                =>  FormattingLogger(loggerFactory.logger(
+                        category, `FormattingLogger`));"
+    shared formal
+    Logger logger<Wrapper>(
+            Category category,
+            "The class that will be used to wrap the returned
+             value, if any. This may be used by loggers that
+             log source file and line number information."
+            ClassOrInterface<Wrapper>? wrapper=null)
+            given Wrapper satisfies Object;
+}
```

Comment:
@jvasileff I don't understand what motivated you to introduce a whole single-method factory interface here.

To me it looks like this alias works as well:

```ceylon
shared alias LoggerFactory => Logger(Category, ClassOrInterface<out Object>=);
```

Am I wrong?